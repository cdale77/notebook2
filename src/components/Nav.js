import React from "react";
import PropTypes from "prop-types";
import UserMenu from "./UserMenu";

const LargeNav = ({ signedIn, signOut, clearCurrentBook }) => {
  const userMenu = () => {
    if (signedIn) {
      return <UserMenu signOut={signOut} clearCurrentBook={clearCurrentBook} />;
    }
  };
  return (
    <nav className="nav">
      <div className="logo">NoteBook2</div>
      {userMenu()}
    </nav>
  );
};

LargeNav.propTypes = {
  clearCurrentBook: PropTypes.func,
  signOut: PropTypes.func.isRequired,
  signedIn: PropTypes.bool
};

export default LargeNav;
