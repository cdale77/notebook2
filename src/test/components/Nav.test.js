import React from "react";
import Renderer from "react-test-renderer";
import Nav from "../../components/Nav";

describe("Nav", () => {
  describe("rendering", () => {
    describe("logged in", () => {
      it("should render the component with the UserMenu", () => {
        const component = Renderer.create(
          <Nav signedIn={true} signOut={jest.fn()} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe("logged out", () => {
      it("should render the component without the UserMenu", () => {
        const component = Renderer.create(
          <Nav signedIn={false} signOut={jest.fn()} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
