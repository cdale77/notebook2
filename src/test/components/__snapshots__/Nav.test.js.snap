// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav rendering logged in should render the component with the UserMenu 1`] = `
<nav
  className="nav"
>
  <div
    className="logo"
  >
    NoteBook2
  </div>
  <div
    className="user-menu"
  >
    <span
      className="clear-current-book-link"
      onClick={[Function]}
    >
      All NoteBooks
    </span>
    <span
      className="signout-link"
      onClick={[Function]}
    >
      Log Out
    </span>
  </div>
</nav>
`;

exports[`Nav rendering logged out should render the component without the UserMenu 1`] = `
<nav
  className="nav"
>
  <div
    className="logo"
  >
    NoteBook2
  </div>
</nav>
`;
